core/src/main/java/com/eaglegenomics/simlims/core/SecurityProfile.java

```diff
@@ -31,16 +41,30 @@
   @Id
   @GeneratedValue(strategy = GenerationType.AUTO)
   private Long profileId = SecurityProfile.UNSAVED_ID;
-  @ManyToOne
+  @ManyToOne(targetEntity = UserImpl.class)
+  @JoinColumn(name = "owner_userId")
   private User owner = null;
-  @ManyToMany
-  private Collection<User> readUsers = new HashSet<User>();
-  @ManyToMany
-  private Collection<User> writeUsers = new HashSet<User>();
-  @ManyToMany
-  private Collection<Group> readGroups = new HashSet<Group>();
-  @ManyToMany
-  private Collection<Group> writeGroups = new HashSet<Group>();
+
+  @ManyToMany(targetEntity = UserImpl.class)
+  @JoinTable(name = "SecurityProfile_ReadUser", joinColumns = { @JoinColumn(name = "readUser_userId") }, inverseJoinColumns = {
```

Comment:
joinColumns and inverseJoinColumns are backwards - joinColumns are the columns linking to `this`, inverse link to the other entity (User in this case)

Same deal with the next 3 `@JoinTable`s